src/components/Deliverable/DeliverableTrackline.tsx

```diff
@@ -128,9 +127,13 @@ function DeliverableTrackLine(props: DeliverableTargetLineProps) {
 					donors={donors}
 					TracklineId={data.updateDeliverableTrackingLineitemDetail.id}
 					TracklineFyId={data.updateDeliverableTrackingLineitemDetail.financial_year?.id}
-					data={donorFormData}
+					data={Object.keys(donorFormData).length ? donorFormData : {}}
```

Comment:
this can be simplified if you just give a default value to `data` in  `DeliverableTracklineDonorYearTags` component
like this 
`function DeliverableTracklineDonorYearTags({ data= {} })`

Reply:
it stores dynamic key fields and its not required at the time of creation thats why i used keys length

Reply:
You are passing `{}` . It can be passed as a default value right?